plugins/default/lib/component_health.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+import sys
+import json
+import argparse
+import urllib2
+
+import nagios
+
+class RequestError(Exception):
+    def __init__(self, message):
+        self.message = message
+
+def generate_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-H","--host", action="store", required=True, help="Host")
+    parser.add_argument('-P','--port', action="store", required=True, help="Port")
+    parser.add_argument("-E","--endpoint", action="store", required=False, help="Endpoint", default="/sys/info/health")
+    return parser
+
+def do_request(host, port, endpoint):
+    url =  'http://%s:%s%s' % (host,port,endpoint)
+    try:
+        data = urllib2.urlopen(url).read()
+    except urllib2.HTTPError, error:
```

Comment:
In "modern Python", the `except klass as e:` syntax is normally preferred over the `,`, for that old syntax is often associated with mistakes when we list multiple exception classes like `except (KeyError, ValueError):`.

https://docs.python.org/2/tutorial/errors.html#handling-exceptions